controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go

```diff
@@ -70,6 +70,12 @@ type KubeadmControlPlaneSpec struct {
 	// to use for initializing and joining machines to the control plane.
 	KubeadmConfigSpec bootstrapv1.KubeadmConfigSpec `json:"kubeadmConfigSpec"`
 
+	// RolloutBefore is a field to indicate a rollout should be performed
+	// if the specified criteria is met.
+	//
+	// +optional
+	RolloutBefore *RolloutBefore `json:"rolloutBefore,omitempty"`
```

Comment:
I wonder if we should set this field in an e2e test. Of course it shouldn't do anything, but testing the case where nothing should happen is better than not testing it at all in e2e. (e.g. the quickstart test)

Reply:
done.